### PR TITLE
added checkdmarc analyzer

### PR DESCRIPTION
**Pull request by goodlandsecurity**
#145

### CODE REVIEW

**Comment by goodlandsecurity** [comment]
i think i need some help with why the checks are failing as they don't appear to be from the files I added/edited. 

i made sure to follow the guide here: https://intelowl.readthedocs.io/en/latest/Contribute.html#create-a-pull-request

**Comment by Eshaan7** [comment]
@goodlandsecurity Thank you for your contribution. It's only a linter check that's failing. Please fix the merge conflicts and then run this `black . --exclude "migrations|venv"` command.

File: api_app/script_analyzers/observable_analyzers/checkdmarc.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+            dmarc_info = stdout.decode("utf-8"), stderr
+
+            dmarc_str = dmarc_info[0]
+
+            dmarc_json = json.loads(dmarc_str)
+
+            return dmarc_json
+
+        except OSError as e:
```
Comment (Eshaan7) [inline api_app/script_analyzers/observable_analyzers/checkdmarc.py:31]: The try-catch block here is redundant. The `run` function is itself wrapped into a try-catch block inside the `ObservableAnalyzer` class.

File: api_app/script_analyzers/observable_analyzers/checkdmarc.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        try:
+            process = subprocess.Popen(
+                [self.check_command, self.observable_name],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            process.wait()
+            stdout, stderr = process.communicate()
+
+            dmarc_info = stdout.decode("utf-8"), stderr
```
Comment (Eshaan7) [inline api_app/script_analyzers/observable_analyzers/checkdmarc.py:23]: A better way would be,
```
if stderr:
  raise Exception(stderr.decode("utf-8"))
dmarc_str = stdout.decode("utf-8")
```

**Review by Eshaan7 — CHANGES_REQUESTED** [review]
Really glad you followed all steps as given in the docs. Just a few things I'd like to change..

**Comment by goodlandsecurity** [comment]
Of course! This is only the second project I've gotten to contribute to, but I am hopeful to add a few more analyzers though! I removed the try-catch block and used shutil to check if the required binary exists before running the analyzer

**Review by Eshaan7 — APPROVED** [review]

**Comment by Eshaan7** [comment]
Thank you for your contribution! The checkdmarc analyzer will be available with the next release which we will drop soon.

**Pull request merged**